torchtune/utils/_generation.py

```diff
@@ -159,7 +159,7 @@ def generate(
         tokens = custom_generate_next_token(
             model,
             input_pos=curr_input_pos,
-            x=tokens,
```

Comment:
Can you help me understand why we need to do .clone? I worry about performance issues

Reply:
Sure, without it we're running into this error:
```
RuntimeError: Error: accessing tensor output of CUDAGraphs that has been overwritten by a subsequent run. Stack trace: File "/home/mreso/torchtune/torchtune/utils/_generation.py", line 48, in generate_next_token
    return sample(logits, temperature, top_k)
  File "/home/mreso/torchtune/torchtune/utils/_generation.py", line 34, in sample
    return multinomial_sample_one(probs)
  File "/home/mreso/torchtune/torchtune/utils/_generation.py", line 16, in multinomial_sample_one
    return torch.argmax(probs / q, dim=-1, keepdim=True).to(dtype=torch.int). To prevent overwriting, clone the tensor outside of torch.compile() or call torch.compiler.cudagraph_mark_step_begin() before each model invocation.
```
Could not get this to work with torch.compiler.cudagraph_mark_step_begin() in the first try so I opted for cloning. Can retry to make cudagraph_mark_step_begin() work,  if you think performance will take a big hit.